.github/workflows/goreleaser.yaml

```diff
@@ -0,0 +1,56 @@
+name: release
+on:
+  pull_request:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
+      - name: Get the image tag
+        run: |
+          # Source: https://github.community/t/how-to-get-just-the-tag-name/16241/32
+          if [[ $GITHUB_REF == refs/tags/* ]]; then
+            echo IMAGE_TAG="${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+          fi
+
+      - name: Docker Login
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_USERNAME }}
+          password: ${{ secrets.QUAY_PASSWORD }}
```

Comment:
This will likely fail on PRs since PRs do not have access to repo secrets.
Probably need to add an `if` expression to only run on tag pushes.

Reply:
TIL about `if: startsWith(github.ref, 'refs/tags')` which seems like it should do the trick.